Tasks/DotnetCLIRestore/restorecommand.ts

```diff
@@ -0,0 +1,306 @@
+        const authInfo = new auth.NuGetExtendedAuthInfo(null, externalAuthArr);
+
+        // Setting up sources, either from provided config file or from feed selection
+        tl.debug('Setting up sources');
+
+        // If there was no nuGetConfigPath, NuGetConfigHelper will create one
+        const nuGetConfigHelper = new NuGetConfigHelper2(
+            null,
+            getTempNuGetConfigPath(),
+            authInfo,
+            { credProviderFolder: null, extensionsDisabled: true },
+            null /* tempConfigPath */,
+            false /* useNugetToModifyConfigFile */);
+
+        let credCleanup = () => { return; };
+
+        const sources: Array<auth.IPackageSource> = new Array<auth.IPackageSource>();
+        let feedUri: string = "https://nuget.pkg.github.com/" + username + "/index.json";
+        sources.push(<auth.IPackageSource>
+            {
+                feedName: "github",
+                feedUri: feedUri,
+                isInternal: false
+            });
+
+        // Creating NuGet.config for the user
+        if (sources.length > 0) {
+            tl.debug(`Adding the following sources to the config file: ${sources.map(x => x.feedName).join(';')}`);
+            nuGetConfigHelper.addSourcesToTempNuGetConfig(sources);
+            credCleanup = () => {
+                tl.rmRF(nuGetConfigHelper.tempNugetConfigPath);
+            };
+        } else {
+            tl.debug('No sources were added to the temp NuGet.config file');
+        }
+
+        // Setting creds in the temp NuGet.config if needed
+        nuGetConfigHelper.setAuthForSourcesInTempNuGetConfig();
+        const configFile = nuGetConfigHelper.tempNugetConfigPath;
+        nuGetConfigHelper.backupExistingRootNuGetFiles();
+        const dotnetPath = tl.which('dotnet', true);
+
+        let projectFiles = CreateProjectFiles(configFile);
+
+        try {
+            for (const projectFile of projectFiles) {
+                await dotnetAddAsync(dotnetPath, projectFile, packageName, packageVersion, configFile);
+                await dotNetRestoreAsync(dotnetPath, projectFile, packagesDirectory, packageVersion, configFile, noCache, verbosity);
+            }
+        } finally {
+            credCleanup();
+            nuGetConfigHelper.restoreBackupRootNuGetFiles();
+        }
+
+        tl.setResult(tl.TaskResult.Succeeded, tl.loc('PackagesInstalledSuccessfully'));
+
+    } catch (err) {
+        tl.error(err);
+        if (buildIdentityDisplayName || buildIdentityAccount) {
+            tl.warning(tl.loc('BuildIdentityPermissionsHint', buildIdentityDisplayName, buildIdentityAccount));
+        }
+
+        tl.setResult(tl.TaskResult.Failed, tl.loc('PackageFailedToInstall'));
+    }
+}
+
+function getTempNuGetConfigPath(): string {
+    const tempNuGetConfigBaseDir = tl.getVariable("Agent.BuildDirectory") || tl.getVariable("Agent.TempDirectory");
```

Comment:
Nitpick: Should we rather prioritize use us tempDirectory since the file name here is NuGet.config

Reply:
Keeping it as it is.